program_transformers/src/token_extensions/mint.rs

```diff
@@ -0,0 +1,306 @@
+        .await
+        .map_err(|e| ProgramTransformerError::DatabaseError(e.to_string()))?;
+    Ok(())
+}
+
+async fn upsert_asset(
+    m: &MintAccount,
+    key_bytes: Vec<u8>,
+    slot: i64,
+    db: &DatabaseConnection,
+    txn: &DatabaseTransaction,
+) -> ProgramTransformerResult<()> {
+    let is_nft = is_token_nft(m);
+    let owner_type = match is_nft {
+        true => OwnerType::Single,
+        false => OwnerType::Token,
+    };
+    if is_nft {
+        let token_account: Option<token_accounts::Model> = find_model_with_retry(
+            db,
+            "token_accounts",
+            &token_accounts::Entity::find()
+                .filter(token_accounts::Column::Mint.eq(key_bytes.clone()))
+                .filter(token_accounts::Column::Amount.gt(0))
+                .order_by(token_accounts::Column::SlotUpdated, Order::Desc),
+            RETRY_INTERVALS,
+        )
+        .await
+        .map_err(|e| ProgramTransformerError::AssetIndexError(e.to_string()))?;
+
+        match token_account {
+            Some(ta) => {
+                upsert_assets_token_account_columns(
+                    AssetTokenAccountColumns {
+                        mint: key_bytes.clone(),
+                        owner: Some(ta.owner),
+                        frozen: ta.frozen,
+                        delegate: ta.delegate,
+                        slot_updated_token_account: Some(ta.slot_updated),
+                    },
+                    txn,
+                )
+                .await?
+            }
+            None => {
+                if m.account.supply == 1 {
+                    warn!(
+                        target: "Account not found",
+                        "Token acc not found in 'token_accounts' table for mint {}",
+                        bs58::encode(&key_bytes).into_string()
+                    );
+                }
+            }
+        }
+
+        let extensions = serde_json::to_value(m.extensions.clone())
+            .map_err(|e| ProgramTransformerError::SerializatonError(e.to_string()))?;
+
+        let class = match is_nft {
+            true => SpecificationAssetClass::Nft,
+            false => SpecificationAssetClass::FungibleToken,
+        };
+
+        let asset_model = asset::ActiveModel {
+            id: Set(key_bytes.clone()),
+            owner_type: Set(owner_type),
+            supply: Set(m.account.supply as i64),
+            supply_mint: Set(Some(key_bytes.clone())),
+            specification_version: Set(Some(SpecificationVersions::V1)),
+            specification_asset_class: Set(Some(class)),
+            nonce: Set(Some(0)),
+            seq: Set(Some(0)),
+            compressed: Set(false),
+            compressible: Set(false),
+            asset_data: Set(Some(key_bytes.clone())),
+            slot_updated: Set(Some(slot)),
+            burnt: Set(false),
+            mint_extensions: Set(Some(extensions)),
+            ..Default::default()
+        };
```

Comment:
Could this be moved down to be above the SQL execution?  Also, I don't think you will update mint_extensions in the OnConflict section. Is that intentional?